wp-content/themes/core/components/text/text.php

```diff
@@ -9,7 +9,7 @@
 // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
 $c = Text_Controller::factory( $args );
 
-if ( empty( $c->get_content() ) ) {
+if ( $c->get_content() === null ) {
```

Comment:
This is odd, empty should return true for NULL. https://www.php.net/manual/en/function.empty.php#114267 

Should we check for empty OR null to make sure other general empty conditions are covered?

Reply:
Yes, good call. I will add the empty() check back in.

Reply:
I guess the "OR" won't work. The issue we are having is that the string of zero - `"0"` is returning true for being empty. So if we have an empty OR null, it still won't pass.

Reply:
Could we cast the value to an int prior the check if we always assume it's a number?

Reply:
Well it's technically a text field, so it can be anything (can have a percentage sign or I assume it could have a string value if they wanted to)

Reply:
I could explicitly just check that it's not equal to a string of 0, since that is the only issue we are having.

Reply:
`if ( empty( $c->get_content() ) || $c->get_content() !== '0' )`

Reply:
Sounds good!

Reply:
> `if ( empty( $c->get_content() ) && $c->get_content() !== '0' )`

This will still fail because of the order that PHP processes statements (left to right). 

You'll need to reverse the check to check for the '0' string first.

Reply:
Actually, scratch that. This will work well.